docs/proposals/multi-node/README.md

```diff
@@ -0,0 +1,623 @@
+While this has the advantage of avoid a dependency on an external component like [kupid](https://github.com/gardener/kupid), the disadvantage is that such built-in support necessarily limits what kind of topologies of the underlying cluster will be supported.
+Hence, it is better to keep `etcd-druid` altogether agnostic of issues related to scheduling and cluster-topology.
+
+### PodDisruptionBudget
+
+This proposal recommends that `etcd-druid` should deploy [`PodDisruptionBudget`](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets) (`maxUnavailable` set to `floor(<cluster size>/2)`) for multi-node etcd clusters to ensure that any planned disruptive operations can try and honour the disruption budget to ensure high availability of the etcd cluster.
+
+## Rolling updates to etcd members
+
+Any changes to the `Etcd` resource spec that might result in a change to `StatefulSet` spec or otherwise result in a rolling update of member pods should be applied/propagated by `etcd-druid` only when the etcd cluster is fully healthy to reduce the risk of quorum loss during the updates.
+If the cluster is unhealthy, `etcd-druid` must restor it to full health before proceeding with the rolling update.
+This can be further optimized in the future to handle the cases where rolling updates can still be performed on an etcd cluster that is not fully healthy.
+
+## Follow Up
+
+### Shoot Control-Plane Migration
+
+This proposal adds support for multi-node etcd clusters but it should not have significant impact on [shoot control-plane migration](https://github.com/gardener/gardener/blob/master/docs/proposals/07-shoot-control-plane-migration.md) any more than what already present in the single-node etcd cluster scenario.
+But to be sure, this needs to be discussed further.
+
+### Performance impact of multi-node etcd clusters
+
+Multi-node etcd clusters incur a cost on [write performance](https://etcd.io/docs/v2/admin_guide/#optimal-cluster-size) as compared to single-node etcd clusters.
+This performance impact needs to be measured and documented.
+
+### Metrics, Dashboards and Alerts
+
+There already metrics exported by etcd and `etcd-backup-restore` which are visualized in monitoring dashboards and also used in triggering alerts.
+These might have hidden assumptions about single-node etcd clusters.
+These might need to be enhanced and potentially new metrics, dashboards and alerts configured to cover the multi-node etcd cluster scenario.
+
+### Costing
+
+Multi-node etcd clusters will clearly involve higher cost (when compared with single-node etcd clusters) just going by the CPU and memory usage for the additional members.
+On top of that, the [different options](#data-persistence) for persistence for etcd data for the members will have different cost implications.
+Such cost impact needs to be assessed and documented.
```

Comment:
...if nothing else then to help understand what are the cost implications of ETCD singletons vs. Clustered ETCDs.